pkg/controller/statusmanager/status_manager.go

```diff
@@ -37,6 +36,8 @@ type StatusManager struct {
 	name    string
 	version string
 
+	managerMutex sync.Mutex
```

Comment:
The golang convention when you have a single mutex that covers the whole object is to use an "embedded field", normally at the top of the struct. So:

    type StatusManager struct {
            sync.Mutex

            client client.Client
            ...

and then you just use `status.Lock()` / `defer status.Unlock()`